nemo/collections/asr/parts/utils/transcribe_utils.py

```diff
@@ -11,15 +11,295 @@
+        else:
+            for hyp, ref in zip(hyps, refs):
+                print("hyp:", hyp)
+                print("ref:", ref)
+
+    wrapped_hyps = wrap_transcription(hyps)
+    return wrapped_hyps
+
+
+def get_buffered_pred_feat(
+    asr: FrameBatchASR,
+    frame_len: float,
+    tokens_per_chunk: int,
+    delay: int,
+    preprocessor_cfg: DictConfig,
+    model_stride_in_secs: int,
+    device: Union[List[int], int],
+    manifest: str = None,
+    filepaths: List[list] = None,
+) -> List[rnnt_utils.Hypothesis]:
+    """
+    Moved from examples/asr/asr_chunked_inference/ctc/speech_to_text_buffered_infer_ctc.py
+    Write all information presented in input manifest to output manifest and removed WER calculation.
+    """
+    # Create a preprocessor to convert audio samples into raw features,
+    # Normalization will be done per buffer in frame_bufferer
+    # Do not normalize whatever the model's preprocessor setting is
+    preprocessor_cfg.normalize = "None"
+    preprocessor = nemo_asr.models.EncDecCTCModelBPE.from_config_dict(preprocessor_cfg)
+    preprocessor.to(device)
+    hyps = []
+    refs = []
+
+    if filepaths and manifest:
+        raise ValueError("Please select either filepaths or manifest")
+    if filepaths is None and manifest is None:
+        raise ValueError("Either filepaths or manifest shoud not be None")
+
+    if filepaths:
+        for l in tqdm(filepaths, desc="Sample:"):
+            asr.reset()
+            asr.read_audio_file(l, delay, model_stride_in_secs)
+            hyp = asr.transcribe(tokens_per_chunk, delay)
+            hyps.append(hyp)
+    else:
+        with open(manifest, "r") as mfst_f:
+            for l in tqdm(mfst_f, desc="Sample:"):
+                asr.reset()
+                row = json.loads(l.strip())
+                if 'text' in row:
+                    refs.append(row['text'])
+                # do not support partial audio
+                asr.read_audio_file(row['audio_filepath'], delay, model_stride_in_secs)
+                hyp = asr.transcribe(tokens_per_chunk, delay)
+                hyps.append(hyp)
+
+    if os.environ.get('DEBUG', '0') in ('1', 'y', 't'):
+        if len(ref) == 0:
+            print("ground-truth text does not present!")
+            for hyp in hyps:
+                print("hyp:", hyp)
+        else:
+            for hyp, ref in zip(hyps, refs):
+                print("hyp:", hyp)
+                print("ref:", ref)
+
+    wrapped_hyps = wrap_transcription(hyps)
+    return wrapped_hyps
+
+
+def wrap_transcription(hyps: List[str]) -> List[rnnt_utils.Hypothesis]:
+    """ Wrap transcription to the expected format in func write_transcription """
+    wrapped_hyps = []
+    for hyp in hyps:
+        hypothesis = rnnt_utils.Hypothesis(score=0.0, y_sequence=[], text=hyp)
+        wrapped_hyps.append(hypothesis)
+    return wrapped_hyps
+
+
+def setup_gpu(cfg: DictConfig) -> Tuple[Union[List[int], int], str, torch.device]:
```

Comment:
Not needed, revert, this should not be a utility method, it should be part of each script.

Reply:
make sense. but this has been replicated in three scripts.

Reply:
Its very generic, hiding it inside function makes it harder to modify for one script vs other

Reply:
I'm okay with reverting it but just don't know in which condition we need to change this for one script and not others?

Reply:
Don't keep it any of the scripts. Managing torch is not a utility functions job.

Reply:
reverted